### PR TITLE
assistant: Remove `low_speed_timeout`

### LINKED ISSUE #19509: Zed Ollama integration is BROKEN starting with 0.158.0
### Check for existing issues

- [X] Completed

### Describe the bug / provide steps to reproduce it

Starting in v 0.158.0, zed preview truncates ollama assistant model output after about a page-worth of text and throws this error:

SCR-20241021-gbdj

`Error interacting with language model error decoding response body`

The error occurs independently of which (of several) different ollama models are chosen and also independently of the prompt. The ollama instance that I am connecting to is on another host on my LAN and is at version 0.3.14.

There is absolutely nothing relevant logged into the zed log.

Downgrading Zed preview to version 0.157.4 fixes this issue with no changes to settings and the exact same prompt.

### Environment

Zed: v0.158.0 (Zed Preview)
OS: macOS 14.7.0
Memory: 64 GiB
Architecture: x86_64

### If applicable, add mockups / screenshots to help explain present your vision of the feature

_No response_

### If applicable, attach your Zed.log file to this issue.

<details><summary>Zed.log</summary><pre>

</pre></details>

EDIT: below are my zed assistant settings

````
  "assistant": {
    "default_model": {
      "provider": "ollama",
      "model": "llama3.1:8b-instruct-q8_0"
    },
    "version": "2"
  },
  "language_models": {
    "ollama": {
      "api_url": "http://pjv-mba:11434",
      "low_speed_timeout_in_seconds": 30
    }
  },
````

### PR DESCRIPTION
**Pull request by mrnugget**
This removes the `low_speed_timeout` setting from all providers as a response to issue #19509.

Reason being that the original `low_speed_timeout` was only as part of #9913 because users wanted to _get rid of timeouts_. They wanted to bump the default timeout from 5sec to a lot more.

Then, in the meantime, the meaning of `low_speed_timeout` changed in #19055 and was changed to a normal `timeout`, which is a different thing and breaks slower LLMs that don't reply with a complete response in the configured timeout.

So we figured: let's remove the whole thing and replace it with a default _connect_ timeout to make sure that we can connect to a server in 10s, but then give the server as long as it wants to complete its response.

Closes #19509

Release Notes:

- Removed the `low_speed_timeout` setting from LLM provider settings, since it was only used to _increase_ the timeout to give LLMs more time, but since we don't have any other use for it, we simply remove the setting to give LLMs as long as they need.

### CODE REVIEW

**Review by KhazAkar — COMMENTED** [review]
Looks okay to my untrained rust eye :)

**Comment by rgbkrk** [comment]
> make sure that we can connect to a server in 10s, but then give the server as long as it wants to complete its response

YES! That's exactly what we need.

File: crates/reqwest_client/src/reqwest_client.rs [hunk 1]
```diff
@@ -31,19 +36,14 @@ impl ReqwestClient {
...
         Ok(client.into())
     }
 
     pub fn proxy_and_user_agent(proxy: Option<http::Uri>, agent: &str) -> anyhow::Result<Self> {
         let mut map = HeaderMap::new();
         map.insert(http::header::USER_AGENT, HeaderValue::from_str(agent)?);
-        let mut client = reqwest::Client::builder()
-            .use_rustls_tls()
-            .default_headers(map);
+        let mut client = Self::builder().default_headers(map);
```
[the remark this answers is not in the document]
Comment (mrnugget) [reply crates/reqwest_client/src/reqwest_client.rs:46]: Just moved to the `Self::builder` method

screenshot-2024-11-15-07 36 55@2x

[the remark this answers is not in the document]
Comment (KhazAkar) [reply crates/reqwest_client/src/reqwest_client.rs:46]: Silly me, sorry 😁

**Pull request merged**